SimDataFormats/GeneratorProducts/src/HepMC3Product.cc

```diff
@@ -17,37 +17,35 @@ using namespace edm;
 using namespace std;
 
 HepMC3Product::HepMC3Product(HepMC3::GenEvent* evt)
-    : evt_(evt), isVtxGenApplied_(false), isVtxBoostApplied_(false), isPBoostApplied_(false) {}
+    : isVtxGenApplied_(false), isVtxBoostApplied_(false), isPBoostApplied_(false) {
+  addHepMCData(evt);
+}
 
 HepMC3Product::~HepMC3Product() {
-  delete evt_;
-  evt_ = nullptr;
   isVtxGenApplied_ = false;
   isVtxBoostApplied_ = false;
   isPBoostApplied_ = false;
```

Comment:
There is no reason to do these assignments in a destructor. I'd suggest either
```cpp
HepMC3Product::~HepMC3Product() = default;
```
or
```cpp
HepMC3Product::~HepMC3Product() {};
```